{"uid":"d21aae50a62481f6","name":"test_apply_expired_coupon_neg","fullName":"tests.backend.coupons.test_apply_coupons#test_apply_expired_coupon_neg","historyId":"97a572918779b9751b68903cf8e77178","time":{"start":1758007412920,"stop":1758007412920,"duration":0},"description":"Verify that applying an expired coupon to an order fails.\n\nCreates an expired coupon, attempts to apply it to a new order,\nand checks that the API returns the correct error and that the coupon\nis recognized as invalid.\n\nArgs:\n    apply_coupon_setup (dict): Fixture with DAOs, API helpers, and test data.\n\nAsserts:\n    - Coupon is expired.\n    - API response matches expected error for expired coupons.\n","descriptionHtml":"<p>Verify that applying an expired coupon to an order fails.</p>\n<p>Creates an expired coupon, attempts to apply it to a new order,<br />\nand checks that the API returns the correct error and that the coupon<br />\nis recognized as invalid.</p>\n<p>Args:<br />\napply_coupon_setup (dict): Fixture with DAOs, API helpers, and test data.</p>\n<p>Asserts:<br />\n- Coupon is expired.<br />\n- API response matches expected error for expired coupons.</p>\n","status":"broken","statusMessage":"RuntimeError: 'cryptography' package is required for sha256_password or caching_sha2_password auth methods","statusTrace":"@pytest.fixture(scope=\"module\")\n    def apply_coupon_setup():\n        \"\"\"Fixture to set up a test environment for applying coupons.\n    \n        Returns:\n            dict: Contains DAOs, API helpers, random product and customer,\n                  and lists for tracking created orders and coupons for teardown.\n        \"\"\"\n    \n        products_dao = ProductsDAO()\n        customers_dao = CustomersDAO()\n        coupons_dao = CouponsDAO()\n>       random_product = products_dao.get_random_product_from_db(qty=1)[0]\n\ntests/backend/coupons/test_apply_coupons.py:35: \n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ \nsrc/dao/products_dao.py:40: in get_random_product_from_db\n    rs_sql = self.db_helper.execute_select(sql)\nsrc/utilities/dbUtility.py:31: in execute_select\n    conn = self.create_connection()\nsrc/utilities/dbUtility.py:23: in create_connection\n    connection = pymysql.connect(host=self.host, user=self.creds['db_user'],\n/opt/hostedtoolcache/Python/3.13.7/x64/lib/python3.13/site-packages/pymysql/connections.py:361: in __init__\n    self.connect()\n/opt/hostedtoolcache/Python/3.13.7/x64/lib/python3.13/site-packages/pymysql/connections.py:669: in connect\n    self._request_authentication()\n/opt/hostedtoolcache/Python/3.13.7/x64/lib/python3.13/site-packages/pymysql/connections.py:979: in _request_authentication\n    auth_packet = _auth.caching_sha2_password_auth(self, auth_packet)\n/opt/hostedtoolcache/Python/3.13.7/x64/lib/python3.13/site-packages/pymysql/_auth.py:267: in caching_sha2_password_auth\n    data = sha2_rsa_encrypt(conn.password, conn.salt, conn.server_public_key)\n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ \n\npassword = b'SpecialStudentsSecret'\nsalt = b'|,iO}P\\x187%\\x1f+0\\x0c\\x01H[\\x02tZl'\npublic_key = b'-----BEGIN PUBLIC KEY-----\\nMIIBIjANBgkqhkiG9w0BAQEFAAOCAQ8AMIIBCgKCAQEAmE7rN2b1SK+cHlbBJuhy\\n7WnyqChNyu7BrMutBWtlMO...EozxYH9BW+KJ23\\nvx4a0Hrs6Ui6XNKiYmsK9l/PwEwYt3phoWR0RAh27OM12+mxaNwalSUs0uosqNFr\\nNwIDAQAB\\n-----END PUBLIC KEY-----\\n'\n\n    def sha2_rsa_encrypt(password, salt, public_key):\n        \"\"\"Encrypt password with salt and public_key.\n    \n        Used for sha256_password and caching_sha2_password.\n        \"\"\"\n        if not _have_cryptography:\n>           raise RuntimeError(\n                \"'cryptography' package is required for sha256_password or\"\n                + \" caching_sha2_password auth methods\"\n            )\nE           RuntimeError: 'cryptography' package is required for sha256_password or caching_sha2_password auth methods\n\n/opt/hostedtoolcache/Python/3.13.7/x64/lib/python3.13/site-packages/pymysql/_auth.py:144: RuntimeError","flaky":false,"newFailed":false,"newBroken":false,"newPassed":false,"retriesCount":0,"retriesStatusChange":false,"beforeStages":[{"name":"apply_coupon_setup","time":{"start":1758007412600,"stop":1758007412730,"duration":130},"status":"broken","statusMessage":"RuntimeError: 'cryptography' package is required for sha256_password or caching_sha2_password auth methods\n","statusTrace":"  File \"/opt/hostedtoolcache/Python/3.13.7/x64/lib/python3.13/site-packages/pluggy/_callers.py\", line 38, in run_old_style_hookwrapper\n    res = yield\n          ^^^^^\n  File \"/opt/hostedtoolcache/Python/3.13.7/x64/lib/python3.13/site-packages/pluggy/_callers.py\", line 139, in _multicall\n    teardown.throw(exception)\n    ~~~~~~~~~~~~~~^^^^^^^^^^^\n  File \"/opt/hostedtoolcache/Python/3.13.7/x64/lib/python3.13/site-packages/_pytest/setuponly.py\", line 36, in pytest_fixture_setup\n    return (yield)\n            ^^^^^\n  File \"/opt/hostedtoolcache/Python/3.13.7/x64/lib/python3.13/site-packages/pluggy/_callers.py\", line 121, in _multicall\n    res = hook_impl.function(*args)\n  File \"/opt/hostedtoolcache/Python/3.13.7/x64/lib/python3.13/site-packages/_pytest/fixtures.py\", line 1140, in pytest_fixture_setup\n    result = call_fixture_func(fixturefunc, request, kwargs)\n  File \"/opt/hostedtoolcache/Python/3.13.7/x64/lib/python3.13/site-packages/_pytest/fixtures.py\", line 891, in call_fixture_func\n    fixture_result = next(generator)\n  File \"/home/runner/work/Ecommerce-Pytest-Framework/Ecommerce-Pytest-Framework/demostore_automation/tests/backend/coupons/test_apply_coupons.py\", line 35, in apply_coupon_setup\n    random_product = products_dao.get_random_product_from_db(qty=1)[0]\n                     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^\n  File \"/home/runner/work/Ecommerce-Pytest-Framework/Ecommerce-Pytest-Framework/demostore_automation/src/dao/products_dao.py\", line 40, in get_random_product_from_db\n    rs_sql = self.db_helper.execute_select(sql)\n  File \"/home/runner/work/Ecommerce-Pytest-Framework/Ecommerce-Pytest-Framework/demostore_automation/src/utilities/dbUtility.py\", line 31, in execute_select\n    conn = self.create_connection()\n  File \"/home/runner/work/Ecommerce-Pytest-Framework/Ecommerce-Pytest-Framework/demostore_automation/src/utilities/dbUtility.py\", line 23, in create_connection\n    connection = pymysql.connect(host=self.host, user=self.creds['db_user'],\n                                 password=self.creds['db_password'],\n                                 port=self.port)\n  File \"/opt/hostedtoolcache/Python/3.13.7/x64/lib/python3.13/site-packages/pymysql/connections.py\", line 361, in __init__\n    self.connect()\n    ~~~~~~~~~~~~^^\n  File \"/opt/hostedtoolcache/Python/3.13.7/x64/lib/python3.13/site-packages/pymysql/connections.py\", line 669, in connect\n    self._request_authentication()\n    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~^^\n  File \"/opt/hostedtoolcache/Python/3.13.7/x64/lib/python3.13/site-packages/pymysql/connections.py\", line 979, in _request_authentication\n    auth_packet = _auth.caching_sha2_password_auth(self, auth_packet)\n  File \"/opt/hostedtoolcache/Python/3.13.7/x64/lib/python3.13/site-packages/pymysql/_auth.py\", line 267, in caching_sha2_password_auth\n    data = sha2_rsa_encrypt(conn.password, conn.salt, conn.server_public_key)\n  File \"/opt/hostedtoolcache/Python/3.13.7/x64/lib/python3.13/site-packages/pymysql/_auth.py\", line 144, in sha2_rsa_encrypt\n    raise RuntimeError(\n    ...<2 lines>...\n    )\n","steps":[],"attachments":[],"parameters":[],"shouldDisplayMessage":true,"stepsCount":0,"attachmentsCount":0,"hasContent":true,"attachmentStep":false}],"afterStages":[],"labels":[{"name":"tag","value":"ebe12"},{"name":"tag","value":"applycoupon"},{"name":"parentSuite","value":"tests.backend.coupons"},{"name":"suite","value":"test_apply_coupons"},{"name":"host","value":"runnervmf4ws1"},{"name":"thread","value":"2079-MainThread"},{"name":"framework","value":"pytest"},{"name":"language","value":"cpython3"},{"name":"package","value":"tests.backend.coupons.test_apply_coupons"},{"name":"resultFormat","value":"allure2"}],"parameters":[],"links":[],"hidden":false,"retry":false,"extra":{"severity":"normal","retries":[],"categories":[{"name":"Test defects","matchedStatuses":[],"flaky":false}],"tags":["ebe12","applycoupon"]},"source":"d21aae50a62481f6.json","parameterValues":[]}